staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -342,7 +377,11 @@ func (t *tracker) Delete(gvr schema.GroupVersionResource, ns, name string) error
 			return err
 		}
 		if objMeta.GetNamespace() == ns && objMeta.GetName() == name {
+			obj := t.objects[gvr][i]
 			t.objects[gvr] = append(t.objects[gvr][:i], t.objects[gvr][i+1:]...)
+			if t.watchers[gvr] != nil && t.watchers[gvr][ns] != nil {
+				t.watchers[gvr][ns].Delete(obj)
```

Comment:
The unit test should have failed if I watched across all namespaces and deleted in this one.